ledger/src/blockstore.rs

```diff
@@ -1531,25 +1533,29 @@ impl Blockstore {
         if index0.is_none() {
             return Ok(None);
         }
-        let mut index0 = self.transaction_status_index_cf.get(0)?.unwrap();
+        let mut index0 = index0.unwrap();
         let mut index1 = self.transaction_status_index_cf.get(1)?.unwrap();
 
+        let mut w_active_transaction_status_index =
+            self.active_transaction_status_index.write().unwrap();
         if !index0.frozen && !index1.frozen {
             index0.frozen = true;
```

Comment:
I think if you move both the `active index` and the "max_slot` field into the lock, you can get rid of this database column entirely

Reply:
nvm, from discussion realized there's no efficient way to reconstruct this on reopen, moar columns!